File: app/src/main/java/com/example/tanjinc/myapplication/fragment/testpage/TestPresenter.java
package com.example.tanjinc.myapplication.fragment.testpage;

import com.example.tanjinc.myapplication.bean.TestBean;

import java.util.ArrayList;

public class TestPresenter implements TestContract.Presenter {
    private TestContract.View mView;

    public TestPresenter(TestContract.View view) {
        mView = view;
    }

    @Override
    public void loadTest() {
        ArrayList<TestBean> beanArrayList = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            beanArrayList.add(new TestBean("index: " + i));
        }
        mView.showTestList(beanArrayList, false);
    }

    @Override
    public void loadMoreTest() {
        ArrayList<TestBean> beanArrayList = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            beanArrayList.add(new TestBean("more: " + i));
        }
        mView.showTestList(beanArrayList, true);
    }

    @Override
    public void refresh() {
        ArrayList<TestBean> beanArrayList = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            beanArrayList.add(new TestBean("refresh: " + i));
        }
        mView.showTestList(beanArrayList, false);
    }
}

File: app/src/main/java/com/example/tanjinc/myapplication/fragment/testpage/TestPresenter.kt
package com.example.tanjinc.myapplication.fragment.testpage

import com.example.tanjinc.myapplication.bean.TestBean

import java.util.ArrayList

class TestPresenter(private val mView: TestContract.View) : TestContract.Presenter {

    override fun loadTest() {
        val beanArrayList = ArrayList<TestBean>()
        for (i in 0..9) {
            beanArrayList.add(TestBean("index: $i"))
        }
        mView.showTestList(beanArrayList, false)
    }

    override fun loadMoreTest() {
        val beanArrayList = ArrayList<TestBean>()
        for (i in 0..9) {
            beanArrayList.add(TestBean("more: $i"))
        }
        mView.showTestList(beanArrayList, true)
    }

    override fun refresh() {
        val beanArrayList = ArrayList<TestBean>()
        for (i in 0..9) {
            beanArrayList.add(TestBean("refresh: $i"))
        }
        mView.showTestList(beanArrayList, false)
    }
}
